data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/RecordType.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.oteltrace;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An enum to represent the record data types supported in {@link OTelTraceSource}.
+ * @since 1.4
+ * TODO: remove in 2.0
+ */
+enum RecordType {
+    otlp,
```

Comment:
We should use the Java name conventions of `OTLP` and `EVENT`. 

You can follow a similar pattern as in `EventType`:

https://github.com/opensearch-project/data-prepper/blob/2b9409fdb18aa45e39c71ef71d40357f1d400abd/data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/EventType.java#L35-L37

In your cause, you should use lowercase strings in the enums though:

```
OTLP("otlp"),
EVENT("event")
```

Also, you can get Jackson to deserialize it directly by annotating the `getByName` method with `@JacksonCreator`.

```
@JacksonCreator
public static RecordType getByName(String name)
```